app/services/course/assessment/submission/zip_download_service.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+class Course::Assessment::Submission::ZipDownloadService
+  class << self
+    # Downloads the submissions and zip them.
+    #
+    # @param [CourseUser] course_user The course user downloading the submissions.
+    # @param [Course::Assessment] assessment The assessments to download submissions from.
+    # @param [String] limit The subset of submissions to download.
+    # @return [String] The path to the zip file.
+    def download_and_zip(course_user, assessment, limit)
```

Comment:
What happens to the service if there are no submissions? Understand that front-end tries to prevent this already.